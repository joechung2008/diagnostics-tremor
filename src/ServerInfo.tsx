import {
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableHeaderCell,
  TableRoot,
  TableRow,
} from "@/components/Table";

const ServerInfo: React.FC<ServerInfoProps> = ({
  deploymentId,
  extensionSync,
  hostname,
  nodeVersions,
  serverId,
  uptime,
}) => {
  const items = [
    {
      name: "Hostname",
      value: hostname,
    },
    {
      name: "Uptime",
      value: uptime,
    },
    {
      name: "Server ID",
      value: serverId,
    },
    {
      name: "Deployment ID",
      value: deploymentId,
    },
    {
      name: "Node Versions",
      value: nodeVersions,
    },
    {
      name: "Extension Sync | Total Sync All Count",
      value: extensionSync.totalSyncAllCount,
    },
  ];

  return (
    <TableRoot>
      <Table aria-label="Server Info">
        <TableHead>
          <TableRow>
            <TableHeaderCell>Name</TableHeaderCell>
            <TableHeaderCell>Value</TableHeaderCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {items.map((item) => (
            <TableRow key={item.name}>
              <TableCell>{item.name}</TableCell>
              <TableCell>{item.value}</TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </TableRoot>
  );
};

export default ServerInfo;
