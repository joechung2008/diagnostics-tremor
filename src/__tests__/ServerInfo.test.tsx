import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import ServerInfo from "../ServerInfo";

describe("ServerInfo", () => {
  it("should render correctly", () => {
    const { container } = render(
      <ServerInfo
        deploymentId="dep-123"
        extensionSync={{ totalSyncAllCount: 5 }}
        hostname="localhost"
        nodeVersions="v18.0.0"
        serverId="srv-456"
        uptime={3600}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
