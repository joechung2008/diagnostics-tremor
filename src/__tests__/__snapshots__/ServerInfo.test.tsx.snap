// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ServerInfo > should render correctly 1`] = `
<div>
  <div
    class="w-full overflow-auto whitespace-nowrap"
  >
    <table
      aria-label="Server Info"
      class="w-full caption-bottom border-b border-gray-200 dark:border-gray-800"
      tremor-id="tremor-raw"
    >
      <thead
        class=""
      >
        <tr
          class="[&_td:last-child]:pr-4 [&_th:last-child]:pr-4 [&_td:first-child]:pl-4 [&_th:first-child]:pl-4"
        >
          <th
            class="border-b px-4 py-3.5 text-left text-sm font-semibold text-gray-900 dark:text-gray-50 border-gray-200 dark:border-gray-800"
          >
            Name
          </th>
          <th
            class="border-b px-4 py-3.5 text-left text-sm font-semibold text-gray-900 dark:text-gray-50 border-gray-200 dark:border-gray-800"
          >
            Value
          </th>
        </tr>
      </thead>
      <tbody
        class="divide-y divide-gray-200 dark:divide-gray-800"
      >
        <tr
          class="[&_td:last-child]:pr-4 [&_th:last-child]:pr-4 [&_td:first-child]:pl-4 [&_th:first-child]:pl-4"
        >
          <td
            class="p-4 text-sm text-gray-600 dark:text-gray-400"
          >
            Hostname
          </td>
          <td
            class="p-4 text-sm text-gray-600 dark:text-gray-400"
          >
            localhost
          </td>
        </tr>
        <tr
          class="[&_td:last-child]:pr-4 [&_th:last-child]:pr-4 [&_td:first-child]:pl-4 [&_th:first-child]:pl-4"
        >
          <td
            class="p-4 text-sm text-gray-600 dark:text-gray-400"
          >
            Uptime
          </td>
          <td
            class="p-4 text-sm text-gray-600 dark:text-gray-400"
          >
            3600
          </td>
        </tr>
        <tr
          class="[&_td:last-child]:pr-4 [&_th:last-child]:pr-4 [&_td:first-child]:pl-4 [&_th:first-child]:pl-4"
        >
          <td
            class="p-4 text-sm text-gray-600 dark:text-gray-400"
          >
            Server ID
          </td>
          <td
            class="p-4 text-sm text-gray-600 dark:text-gray-400"
          >
            srv-456
          </td>
        </tr>
        <tr
          class="[&_td:last-child]:pr-4 [&_th:last-child]:pr-4 [&_td:first-child]:pl-4 [&_th:first-child]:pl-4"
        >
          <td
            class="p-4 text-sm text-gray-600 dark:text-gray-400"
          >
            Deployment ID
          </td>
          <td
            class="p-4 text-sm text-gray-600 dark:text-gray-400"
          >
            dep-123
          </td>
        </tr>
        <tr
          class="[&_td:last-child]:pr-4 [&_th:last-child]:pr-4 [&_td:first-child]:pl-4 [&_th:first-child]:pl-4"
        >
          <td
            class="p-4 text-sm text-gray-600 dark:text-gray-400"
          >
            Node Versions
          </td>
          <td
            class="p-4 text-sm text-gray-600 dark:text-gray-400"
          >
            v18.0.0
          </td>
        </tr>
        <tr
          class="[&_td:last-child]:pr-4 [&_th:last-child]:pr-4 [&_td:first-child]:pl-4 [&_th:first-child]:pl-4"
        >
          <td
            class="p-4 text-sm text-gray-600 dark:text-gray-400"
          >
            Extension Sync | Total Sync All Count
          </td>
          <td
            class="p-4 text-sm text-gray-600 dark:text-gray-400"
          >
            5
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
